week3/BusinessCard/app/src/main/java/com/manuflowers/businesscard/SettingsFragment.kt

```diff
@@ -0,0 +1,84 @@
+package com.manuflowers.businesscard
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.Build
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.RadioGroup
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.fragment.app.DialogFragment
+import com.manuflowers.businesscard.utils.Constants.PREFERENCES_MODE
+import com.manuflowers.businesscard.utils.Constants.THEME_MODE_KEY
+import com.manuflowers.businesscard.utils.ThemeMode
+import kotlinx.android.synthetic.main.fragment_settings.*
+
+class SettingsFragment : DialogFragment(), RadioGroup.OnCheckedChangeListener {
+
+    lateinit var sharedPreferences: SharedPreferences
+
+    override fun onStart() {
+        super.onStart()
+        /**
+         * Set width and height to fragment layout
+         * */
+        dialog?.window?.setLayout(
+            ConstraintLayout.LayoutParams.MATCH_PARENT,
+            ConstraintLayout.LayoutParams.WRAP_CONTENT
+        )
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_settings, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        //Initialize interface
+        themeModeRadioGroup.setOnCheckedChangeListener(this)
+
+        sharedPreferences = activity!!.getSharedPreferences(PREFERENCES_MODE, Context.MODE_PRIVATE)
```

Comment:
Avoid force unwrapping at all costs. This can cause a crash which is a bad user experience. There's a reason why in Kotlin a force unwrap has TWO exclamation marks - it's to emphasize not to do this.

One way to write this would be:

```kotlin
activity?.let {
            sharedPreferences = it.getSharedPreferences(PREFERENCES_MODE, Context.MODE_PRIVATE)
        }
```

Reply:
Thanks for the advice